spec/features/admin/payment_method_spec.rb

```diff
@@ -185,6 +185,16 @@
       expect(page).not_to have_content payment_method3.name
     end
 
+    it "shows only the providers of the existing payment methods" do
+      payment_method1
+      payment_method2
+      payment_method3
+
+      visit spree.admin_payment_methods_path
+
+      expect(page).to have_content "Cash/EFT/etc. (payments for which automatic validation is not required)", count: 2
+    end
```

Comment:
I would rather move this test to a lower level. Controller at least although I think it's the perfect use-case for a view test. The thing we're testing isn't worth booting a browser and going through the entire stack IMO

Reply:
I'm trying to create the view spec but I'm getting an error. I will push what I did until now. Could you help me?

![Screen Shot 2020-06-08 at 00 16 08](https://user-images.githubusercontent.com/27960597/83989605-4f6e5d00-a91d-11ea-81da-97c0c2c535c1.png)

Reply:
that's an helper method defined in the parent ResourceController, maybe the controller is not fully loaded for view specs?
maybe you can load the helper in the view spec explicitly?

if it was an helper it would be:
helper Spree::BaseHelper
but this is an helper method, maybe we can at least try to validate this theory by defining the method in the class_eval of the spec:

```
    controller.singleton_class.class_eval do
      def current_ability
        Spree::Ability.new(Spree.user_class.new)
      end

      def new_object_url
          ...
      end
    end

```

I am sorry, this is not very good code, but I think it's worth taking these ugly shortcuts to get more view specs as we go, this is the 5th spec now :tada:

Reply:
I will try!

Reply:
@luisramos0 I tried with your approach! No success

Reply:
![Screen Shot 2020-06-12 at 18 54 02](https://user-images.githubusercontent.com/27960597/84549337-21529980-acde-11ea-927d-fb3ffda5da4b.png)

Reply:
re my approach, can you please push the code you have tried and the error you get?

Reply:
@luisramos0 I tried to define the method inside the `class_eval` block, but it keeps complaining about the undefined method "new_object_url". The same error that I posted when you recommended defining inside the class_eval.

Reply:
![Screen Shot 2020-06-08 at 00 16 08](https://user-images.githubusercontent.com/27960597/84570225-69b39b00-ad62-11ea-8c00-210fdda49ec4.png)

Reply:
I was able to fix stubbing the method "new_object_url". I will try to finish and I will push the spec.

Reply:
Done!

Reply:
I was forgetting to call "helper_method", sorry @luisramos0 ! I updated the PR with the view test.